nox.py

```diff
@@ -94,6 +58,22 @@ def session_reqcheck(session):
         session.run('gcprepotools', command, reqfile)
 
 
+def collect_sample_dirs(start_dir, blacklist=set()):
+    """Recursively collects a list of dirs that contain tests."""
+    # Collect all the directories that have tests in them.
+    samples = []
+    for parent, subdirs, files in os.walk(start_dir):
+        if any(f for f in files if f[-8:] == '_test.py'):
+            # This dir has tests in it. Add it, and stop recursion
+            samples.append(parent)
+            del subdirs[:]
+        else:
+            # Filter out dirs we don't want to recurse into
+            subdirs[:] = [s for s in subdirs
+                          if re.match(r'[a-z]', s) and s not in blacklist]
```

Comment:
more of an opinion thing but I think s[0].isAlpha() is better since it accomplishes the same thing without adding re as a dependency.